includes/REST_API/Stories_Controller.php

```diff
@@ -464,6 +465,11 @@ protected function prepare_links( $post ): array {
 		$links = parent::prepare_links( $post );
 		add_post_type_support( Story_Post_Type::POST_TYPE_SLUG, 'revisions' );
 
+		// Ensure that no data about story locks, leaks publically. 
+		if ( ! $this->get_post_type_cap( $this->post_type, 'edit_posts' ) ) {
+			return $links;
+		}
```

Comment:
Do not embed lock / lock user if you can't edit the post type. This should always have been here.